### DISCUSSION TITLE
Dev Branch 2.2 Log

### DISCUSSION THREAD

**Discussion by MXKhronos** [HelixNebulaStudio/RiseOfTheDead#727] [General]
# Dev Branch 2.2 Log
*(Note: italicized text are deemed likely hypothetical and concept only and may not be developed in the end)*
I was suggested to make a dev update log for what's being developed so far. Note that I don't think much of the current development are testable just yet but it's best to clue you guys in.

## Released
Some changes and new features are already released. Such as the new sliding changes, live pvp kill feed, `skullburstmod`, mission tree's timeline buttons for Rat and Bandit.

- **Sliding Changes**
If you must know, I've been enjoying Deadlock quite a bit. And after experiencing the movement mechanics in the game, I realize sliding in RotD is lacking control. After a couple tweaks and rewrite, you can now turn while sliding and sliding down slopes gives you more speed.
*I want to add more movement mechanics such as vaulting walls, double jumping, wall jumping, etc.. for a new clothing set, but we'll have to see if I can finish modeling the fit. I'd also like to package it into a "Scout" class bundle with some other tbd features.*

- **Live PvP Kill Feed**
If you noticed the new kill feed in Slaughter Fest, that was one of the new live feed implementation that will be used in the currently developed gamemode for halloween.

- **`skullburstmod`**
Intensifying Skull Burst is a new type of passive mod that I want to design for the future of mods and phase out generic value increase mods that just adds dmg, fire rate, reduce reload time etc.. as they are way to plain and limit build variety. More upcoming mods will be designed with more mixed attributes.

- **Mission Tree Timelines**
So there are 3 new buttons on the mission menu by the mission tree, Main, Rat and Bandit as the timeline for rat and bandit are a side plot to the main story. I'm currently working on a Cargo Ship Coop that is tied to the next coming Rat mission.
`Fall of the Living` is a continuation of the main plot from the survivors perspective, while rat and bandit's side plot are from the perspective of the factions respectively.

   Rat: You helped the rats, the bandits lost their infector Stan during `Double Cross`. Revas can now experiment on Stan. The rats have to deal with the increased aggression by the bandits (Cargo Ship Coop).

    Bandits: You helped the bandits, the rats could not capture Stan in `Double Cross`. Zark still needs information on Sector C. The bandits have to deal with the potential espionage and done by the rats. (tbd).

 ## In Development

- **Cargo Ship Coop**
This will be a replayable mission based mode where you escort a R.A.T. cargo crew on their ship to pickup cargo around Wrighton Dale and Wrighton Dale East. However, while on the way to Wrighton Dale East, you get attacked by squads of Bandits. After fighting them off, Skipper will be severely injured and you'll have to drive the cargo ship back to the harbor.

    Lore: After helping the rats in `Double Cross`, the bandits has been upping their aggression. Defend against bandit attacks for the Rats, *help rat shops in safehouses*.
    
    Map: `CargoShip`, so far, development at the enemy spawning stage. There will be different stage to fight off bandits. *Helicopter sniper stage where you have to fight off snipers from the helicopter during a heavy fog*, *Bandits dinghy stage where bandits on dinghy circles the cargo ship firing at rat crew*, *Bandits boarding stage*, *tbd*

   Currently on hold for halloween development.

- **Nekron's Gambit**
Inspired by Deadlock of course, it will be a Team PvPvE for now, where you have 2 lanes which you have to fight and push to gain territory and eventually wipe out the other side. Max players is 8 and it'll be more of a MOBA, so TTK is a lot slower. Giving you time to react and respond to what enemies throw at you.
    
    Lore: It is still Tbd, but the concept is that *you have became a pawn in Nekron's game of chess*.
    Map: `NekronsGambit`, so far, the draft layout of the map is complete. Assume all assets are placeholder for now. Currently working on the map's game mechanics.
    
   **Gameplay**:
       - You start with a `p250` with infinite reload and a `survivalknife`.
       - Every minute, a wave of pawns spawn and moves through the lane. When they encounter enemy pawns, they fight each other.
       - Players help kill enemy pawns to push lanes, pawns are required in order to fight Knights, Bishops, Queen and Rooks.
       - Players also fight enemy team players in order to push lanes.
       - When player dies, they respawn after a `tbd respawn timer`, *and pick their spawn location based on how pushed up their lane is.*
       - *Enemy pawns dying near player will give players a currency called `sol`, you can use sols to upgrade your from a `p250` to other weapons in the game.*
       - *You can also use `sols` to purchase character upgrades for more hp, armor, hp regen, sliding distance, jump height etc..*
       - *Random neutral pawns will also be scatter around the map to earn more sols.*

    **Reward**:
        - In a previous post, I want to make this update to add replayability by gameplay rather than rewards. So I wouldn't want to introduce any item rewards here yet, I may however add a achievement for winning once in Nekron's Gambit.

    I hope to complete this game mode within the next week to a playable state in terms of gameplay.

![image](https://github.com/user-attachments/assets/ba1f2c43-9e82-4d07-90b5-942b128b75a3)
![image](https://github.com/user-attachments/assets/4fa035a2-56c0-4a54-ba41-ed8053326e6b)

- **Halloween Update**
This year's halloween update will be slightly smaller than previous halloweens. There won't be a event pass this time.

    With the Slaughter Fest changes, and the addition of the board mission `Slaughter King`:
       - *I want to add a `slaughterfestcandybag` that contains individual type of candies when unboxed such as the `spookymarshmallow`, `zombiejello`, `cherrybloodbar` and `eyeballgummies`. These candies will have a tbd effect when eaten. (open to suggestions)*
       - *Completing `Slaughter King` will drop a `slaughterfestcandybag`.*
       - *As for the Slaughterfest 2024 achievement, I'll add it for getting first place in Slaughterfest.*
       - *As for obtaining previous halloween items, maybe `slaughterfestcandybag` can be exchanged for masks in the cauldron*
       - *Maybe add some other ways to obtain `slaughterfestcandybag` through normal gameplay.*

- **Ninja Set** (halted)
After the sliding changes, I started working on a ninja outfit (Head and Chest) which would give you more movement abilities. But that's currently halted and may resume along the Cargo Ship development.

## Conclusion
Currently the ambition is a bit big and work load seems a lot, I know. I'm not sure exactly how many people wants and enjoy just another event pass to grind but I want a change from just making a bunch of rewards for this event into an event pass. Hence the changes to Slaughterfest from a candy farm to a casual event pvp mode. And why I wanted to change it up with a new pvp mode and smaller halloween event this time and see how it goes. 

Thank you for reading all this which might've taken you at least 5 minutes. If you have feedback or questions, feel free to ask, but since a lot is still in concept limbo, I might not be able to give a definite answer.
~Khronos

**Comment by Leroy8888** [comment]
Most of the things in here seem nice additions. Don't really have a lot of feedback I can currently provide besides some small points.

I think it would be nice addition to allow the candybags to be exchanged for masks or hats, because currently even while it is halloween, there is no way to obtain them. It would be nice to have at least some way to get them. Not all of them have to be included or any new ones, but just a few from the later years. Maybe even the tophat? I noticed it costing a shocking amount when trading with Icarus, so at the same time not sure if it's meant to come back at one point.

PvP in general is a bit of a mode that some people may rather avoid. It puts you up against players with varying skill and such may often seem like the player is either fairly bad or just up against impossible odds. Most games seem to tackle this by giving PvP worthwhile rewards so that players want to participate or spend effort to become better at it. There are also a handfull of games which do not do this, but even so PvP would still grant some benefits which couldn't be achieved via PvE. 

Seems like there wouldn't be any harm in also including these candybags for the Gambit, at least for halloween. I also believe it should have some reward. Basically everything you do in RoTD leads up to the player gaining something. Having a mode that (possibly) doesn't give you anything usefull in game seems a bit odd for that reason. Same reason for SF. If I for example still was new I would rather spend my time on continueing the grind.

However maybe even some specific drops or entities could appear in the Gambit? Bit of vague concept, but it's something along the lines of introducing content that can be found at a pvp zone or world. However such additions may also lock some content behind a certain skill level.

**Comment by MrPigZ** [comment]
Could one of the candies give player low gravity so that they could "fly"?

If you join SF mid-round, you are at a disadvantage when it comes to winning the round, but you can freely leave and find a better server. This doesn't really affect other player's gaming experience. How will unbalanced teams and leavers be dealt with in Nekron's Gambit or will the snowballl-effect be so tiny that having a bad team doesn't matter? 

SF is currently almost empty, and one reason is that there are no rewards to gain. Do people want a new pass? People certainly want rewards, but do they want to keep doing the same thing over and over again? Probably not. So, if a new pass is added someday, a new way to earn levels will be needed. A poll to ask the community could be made.

**Comment by Leroy8888** [comment]
It feels like he latest addition to dev branch decreased the speed boost of slide jumps. Was this intentional? Slide jumps in dev from the fence in bloxmarkt to the fence next to office took about 12 - 13 seconds in dev branch, but 9 - 10 in main branch. It's pretty noticable since slide jumping is something players do during the entire time they play the game.

the `ninja shrouds` passive seems a bit situational, especially when it's only for focus weapons. I would like to suggest that it activates the passive for any weapon in ADS. It could also slow the falling speed by default, with either a toggle button for it or without to make a bit more practical in general.

the description of the `ninja cloak` passive is a bit misleading. It states that with it the player is able to climb walls and double jump. While you are able to double jump, you are not able to climb walls. It instead allows the player to perform ledge grabs to boost themselves up at the top of a wall (or a ledge for the matter). When I do think of wall climbing I would assume that you could grab onto a wall then climb towards a certain direction with a set speed, be it with some other tools.

**Reply by MXKhronos** [reply]
>  feels like he latest addition to dev branch decreased the speed boost of slide jumps

I just fixed a bug causing slide speed was affected by the frame rate, so that may be causing different slide and jump distance experience for each player. It should now be fixed but it will feel different to people who used to experience it to slide longer. I'll keep tuning it to player feedback.

> `ninjashroud` passive

Yeah, scrapped that idea. Implementation was also way too tricky.

> `ninjacloak` wall climb description

Now updated to clarify wall ledge climbing.

**Comment by Leroy8888** [comment]
Would it also be an idea to allow some of the upcoming candy effect to be used in SF? They could spawn like medkits and beans and would grant a little offensive (or defensive) boost. Could be nice for if someone decides to camp in the corner with lots of candy with a minigun.

There also was a change made a bit ago where having 300 candy grants you a candy bag if the round ends. This does mean that a player could idle as there is plenty of time (8 x 60 = 480 candies) to get those candies without really doing anything. The cap of 600 candies is also reached fairly quickly when you're actively participating and attacking other player with candy. Could it be possible to set the max to 1000 candies instead? This would create higher stakes and payouts. 

I would also like to give the 1st, 2nd and 3rd place a bit more for actually competing for that spot. Does not have to be anything special, but maybe just +1 candy bag for their respective places? 4, on 1st, 3 on 2nd, ect.

**Reply by MrGarrettFerret** [reply]
What if candy bags were awarded for reaching a certain candy threshold (it should be higher than it currently is, maybe 600-700 range?) and bonus were awarded for placement? It would give players a strong incentive to try and PvP as much as possible, but also allowing those who would prefer to play more passively (e.g. people who don't like PvP) to also have fun and get rewards in the event.

**Reply by MXKhronos** [reply]
There's a chance I won't be implementing any effects to the edible candies anytime soon.

> 300 candy & placement rewards

Yeah, MrPigZ mentioned,
I'll be adding more event board missions to complete in order to gain candybags
